backend/open_webui/utils/tools.py

```
@@ -2,111 +2,203 @@
 import logging
 import re
 import inspect
```

Ruff (0.12.2)

Redefinition of unused `inspect` from line 1

Remove definition: `inspect`

(F811)